packages/editor/src/data/useDataTableColumns.ts

```diff
@@ -0,0 +1,67 @@
+import type { DataTable, DataTableColumn, Variable } from '@axonivy/form-editor-protocol';
+import { useMeta } from '../context/useMeta';
+import { type CheckboxColumn } from '../editor/sidebar/fields/ColumnsCheckboxField';
+import { findAttributesOfType } from './variable-tree-data';
+import { useAppContext } from '../context/AppContext';
+import { createId, useData } from './data';
+import type { BrowserNode } from '@axonivy/ui-components';
+
+export const useDataTableColumns = () => {
+  const { context } = useAppContext();
+  const { element } = useData();
+
+  const variableInfo = useMeta('meta/data/attributes', context, { types: {}, variables: [] }).data;
+  const attributesOfTableType = findAttributesOfType(
+    variableInfo,
+    element && element.id.startsWith('DataTable') ? (element.config as unknown as DataTable).value : ''
+  );
+  const upToDateColumns = element && element.id.startsWith('DataTable') ? (element.config as unknown as DataTable).components : [];
+  const dataTableColumns = convertBrowserNodesToColumns(attributesOfTableType);
+  const boundDataTableColumns: CheckboxColumn[] = dataTableColumns.map(column => ({
```

Comment:
instead of define the type of the variable you can also add a gernic to the map function, then it will be infered:
```suggestion
  const boundDataTableColumns = dataTableColumns.map<CheckboxColumn>(column => ({
```